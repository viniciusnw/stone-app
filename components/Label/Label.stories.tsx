import React from "react";
import { Label, StorybookWrapper } from "@Components";

const LabelDefault = ({color}: any) => (
  <StorybookWrapper>
    <Label label="Default Label" color={color} />
  </StorybookWrapper>
);

export const Default = {
  args: {
    type: "default",
  },
};

export const Primary = {
  args: {
    type: "primary",
  },
};

export const Success = {
  args: {
    type: "success",
  },
};

export const Danger = {
  args: {
    type: "danger",
  },
};

export const Warning = {
  args: {
    type: "warning",
  },
};

export default {
  title: "Label",
  argTypes: {
    type: {
      options: ["default", "primary", "success", "danger", "warning"],
      control: { type: "radio" },
    },
  },
  component: ({ type }: any) =>  (<LabelDefault color={type} />),
};
