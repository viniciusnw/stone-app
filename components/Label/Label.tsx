import React from "react";
import styled from "styled-components/native";
import { View, Text } from "react-native";

const StyledContainer = styled(View)<{ color: string; rounded: boolean }>`
  background-color: ${({ theme, color }) =>
    theme.color[!color ? "default" : color]};
  padding: 5px 10px;
  flex-direction: row;
  align-self: flex-start;
  border-radius: ${({ rounded }) => (rounded ? "50%" : 0)};
`;

const StyledLabel = styled(Text)`
  color: ${({ theme }) => theme.color.black};
  font-weight: 700;
  font-size: 12px;
  color: ${({ theme }) => theme.color.white};
`;

type Props = {
  label: string;
  color: string;
  rounded?: boolean;
};

const Label: React.FC<Props> = ({
  rounded = false,
  label,
  color,
  ...props
}) => {
  return (
    <StyledContainer rounded={rounded} color={color} {...props}>
      <StyledLabel {...props}>{label}</StyledLabel>
    </StyledContainer>
  );
};

export default Label;
