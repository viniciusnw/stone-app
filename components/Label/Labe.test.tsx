import React from "react";
import renderer from "react-test-renderer";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";
import Label from "./Label";

jest.useFakeTimers();

test("Label", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <Label label="Default Label" color='primary' />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
