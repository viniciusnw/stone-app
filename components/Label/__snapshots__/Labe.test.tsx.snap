// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label 1`] = `
<View
  color="primary"
  rounded={false}
  style={
    {
      "alignSelf": "flex-start",
      "backgroundColor": "rgb(10,245,61)",
      "borderBottomLeftRadius": 0,
      "borderBottomRightRadius": 0,
      "borderTopLeftRadius": 0,
      "borderTopRightRadius": 0,
      "flexDirection": "row",
      "paddingBottom": 5,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 5,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 12,
        "fontWeight": "700",
      }
    }
  >
    Default Label
  </Text>
</View>
`;
